src/main/java/io/harness/cf/client/api/CaffeineCache.java

```diff
@@ -5,32 +5,45 @@
 import java.util.ArrayList;
 import java.util.List;
 import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
 
+@Slf4j
 public class CaffeineCache implements Cache {
 
   private final com.github.benmanes.caffeine.cache.Cache<String, Object> cache;
 
   public CaffeineCache(int size) {
     cache = Caffeine.newBuilder().maximumSize(size).build();
+    log.info("CaffeineCache initialized with size {}", size);
   }
 
   @Override
   public void set(@NonNull String key, @NonNull Object value) {
     cache.put(key, value);
+    log.debug("New value in the cache with key {} and value {}", key, value);
   }
 
   @Override
   public Object get(@NonNull String key) {
```

Comment:
This API probably needs to be annotated with @Nullable to signify that it can return null

Reply:
Done